src/ast/ast_enrich.c

```diff
@@ -0,0 +1,247 @@
+#include "ast.h"
+#include "../util/arr.h"
+#include "../util/qsort.h"
+#include "../util/rmalloc.h"
+#include <assert.h>
+
+//------------------------------------------------------------------------------
+//  Annotation context - graph entity naming
+//------------------------------------------------------------------------------
+
+// Compute the number of digits in a non-negative integer.
+static inline int _digit_count(int n) {
+	int count = 0;
+	do {
+		n /= 10;
+		count++;
+	} while(n);
+	return count;
+}
+
+static inline char *_create_anon_alias(int anon_count) {
+	// We need space for "anon_" (5), all digits, and a NULL terminator (1)
+	int alias_len = _digit_count(anon_count) + 6;
+	char *alias = rm_malloc(alias_len * sizeof(char));
+	snprintf(alias, alias_len, "anon_%d", anon_count);
```

Comment:
We name ORDER BY items with a call to `AR_EXP_ToString`, so that routine and this needs to use the same allocator for proper freeing. I opted to go with `rm_malloc` here rather than switching that routine to use malloc, but as such can't use `asprintf`.

Reply:
It's a shame asprintf can't use a custom allocator, it's a shame we have to compute the number of character a number string representation requires, that loop and the use of div...

Reply:
If you prefer, we can calculate string length with `snprintf` like so:
```
int len = snprintf(NULL, 0, "anon_%d", anon_count);
char *alias = rm_malloc(alias_len * sizeof(char));
snprintf(alias, len, "anon_%d", anon_count);
```
Which abstracts away the computation here, but I feel like under the hood is probably slower. I don't like the double-snprintf approach in general, but it is an orthodox approach!